dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -717,10 +719,16 @@ static void AddResponseMessage(ChatCompletionsOptions chatOptions, ChatHistory c
                         streamedContents?.Add(openAIStreamingChatMessageContent);
                         yield return openAIStreamingChatMessageContent;
                     }
+
+                    // Translate all entries into ChatCompletionsFunctionToolCall instances.
+                    toolCalls = OpenAIFunctionToolCall.ConvertToolCallUpdatesToChatCompletionsFunctionToolCalls(
+                        ref toolCallIdsByIndex, ref functionNamesByIndex, ref functionArgumentBuildersByIndex);
+                    // Translate all entries into FunctionCallContent instances for diagnostics purposes.
+                    functionCallContents = ModelDiagnostics.IsSensitiveEventsEnabled() ? toolCalls.Select(this.GetFunctionCallContent).ToArray() : null;
```

Comment:
Do we need to check if we have an `activity`

Reply:
That's implicitly checked. The `ModelDiagnostics` module makes sure that the activity is null when model diagnostics is disabled. And `IsSensitiveEventEnabled` checks if model diagnostics is enabled and the collection of sensitive event is enabled.